### DISCUSSION TITLE
URQL Mutation Auto calls Query

### DISCUSSION THREAD

**Discussion by MrFarhan** [urql-graphql/urql#2506] [Q&A] [answered]
****My scenario:**** I have a different group of radio buttons on click of a button I update the boolean state and enable and disable the radio button state locally and along with that i run the mutation to upadate database.
![image](https://user-images.githubusercontent.com/26528236/174327554-e6d5b6af-faa8-4388-b3df-4c6655578441.png)

The issue is: when i run the mutation frequently like click on radio buttons without any wait so it runs 2 mutation and calls the query but i do not want the query to run.

Issue in short: When i run mutation it runs the query automatically and i don't want that

**Steps to reproduce**

1- Run any mutation and it will call the query it self in background.

**Expected behavior**
1- Should have the ability to control if we need to call the query after mutation call or not.

**Actual behavior**
1- Calling query after every mutation call

**What I found and tried.
* Found about cache-control but where to add this cache I am not calling the query it's being called automatically after the mutation calls.

**Comment by JoviDeCroock** [comment, marked as answer]
As documented the default cache does that https://formidable.com/open-source/urql/docs/basics/document-caching/ an intersection in typename leads to a refetch. You might want a normalized cache https://formidable.com/open-source/urql/docs/graphcache/ which allows doing that in memory
